app/services/ons_data_import/base.rb

```diff
@@ -63,21 +64,27 @@ def set_area_data(location_polygon, geometry, type)
       FROM geom
       WHERE id='#{location_polygon.id}'
     ")
-  end
+    end
 
-  def arcgis_features(offset)
-    params = [
-      "where=1%3D1",
-      "outSR=4326",
-      "f=pgeojson",
-      "outFields=#{name_field}",
-      "resultRecordCount=#{PER_PAGE}",
-      "resultOffset=#{offset * PER_PAGE}",
-    ].join("&")
+    def arcgis_features(offset:, name_field:, api_name:)
+      params = [
+        "where=1%3D1",
+        "outSR=4326",
+        "f=pgeojson",
+        "outFields=#{name_field}",
+        "resultRecordCount=#{PER_PAGE}",
+        "resultOffset=#{offset * PER_PAGE}",
+      ].join("&")
 
-    response = HTTParty.get("#{ARCGIS_BASE_URL}#{api_name}/FeatureServer/0/query?#{params}")
-    raise "Unexpected ArcGIS response: #{response.code}" unless response.success?
+      response = HTTParty.get("#{ARCGIS_BASE_URL}#{api_name}/FeatureServer/0/query?#{params}")
+      # :nocov:
+      raise "Unexpected ArcGIS response: #{response.code}" unless response.success?
+      # :nocov:
 
-    JSON.parse(response.to_s)["features"]
+      response_data = JSON.parse(response.to_s)
+      raise "ArcGIS error: #{response_data}" if response_data.key?("error")
```

Comment:
Just a suggestion. Wouldn't be better to include only the error contents in the message?
```suggestion
      raise "ArcGIS error: #{response_data['error']}" if response_data.key?("error")
```